### ISSUE TITLE
'BigInt' is not assignable to type 'number'.

### ISSUE THREAD

**Issue by fxalgrain** [benhutchins/dyngoose#443]
Hi, 

I got the following error when adding Dyngoose to my project. 

```
node_modules/dyngoose/dist/decorator/attribute-types/number-set.d.ts:8:74 - error TS2344: Type 'NumberSetAttributeMetadata' does not satisfy the constraint 'AttributeMetadata<Value>'.
  Types of property 'manipulateWrite' are incompatible.
    Type '((attributeValue: AttributeValue | null, value: number[] | null, attribute: Attribute<any>) => AttributeValue | null) | undefined' is not assignable to type '((attributeValue: AttributeValue | null, value: Value | null, attribute: Attribute<any>) => AttributeValue | null) | undefined'.
      Type '(attributeValue: AttributeValue | null, value: number[] | null, attribute: Attribute<any>) => AttributeValue | null' is not assignable to type '(attributeValue: AttributeValue | null, value: Value | null, attribute: Attribute<any>) => AttributeValue | null'.
        Types of parameters 'value' and 'value' are incompatible.
          Type 'Value | null' is not assignable to type 'number[] | null'.
            Type 'Value' is not assignable to type 'number[]'.
              Type 'number | BigInt' is not assignable to type 'number'.
                Type 'BigInt' is not assignable to type 'number'.

8 export declare class NumberSetAttributeType extends AttributeType<Value, Metadata> implements IAttributeType<Value> {
```

We can fix the issue in a simple way. The file number-set.metadata.ts should export :
`export interface NumberSetAttributeMetadata extends AttributeMetadata<(number | BigInt)[]> {}`

**Comment by benhutchins** [comment]
@fxalgrain Thank you for the report. Could you confirm the version of TypeScript you're using?

**Comment by fxalgrain** [comment]
I m using the version 4.1.3

**Comment by benhutchins** [comment]
@fxalgrain I just pushed a fix for this, Dyngoose 2.12.2 should resolve it. See 74a8b834b5958616f219199d173cabded8212fac.

**Comment by fxalgrain** [comment]
It works well now. Thanks you !
